pkg/image/disk.go

```diff
@@ -40,47 +40,34 @@ type DiskImage struct {
 	InstallWeakDeps *bool
 }
 
-func NewDiskImage() *DiskImage {
-	return &DiskImage{
-		Base:     NewBase("disk"),
-		PartTool: osbuild.PTSfdisk,
-	}
+type imagePipelineOpts struct {
+	QCOW2Compat string
+	ForceSize   *bool
+	Filename    string
 }
 
-func (img *DiskImage) InstantiateManifest(m *manifest.Manifest,
-	repos []rpmmd.RepoConfig,
-	runner runner.Runner,
-	rng *rand.Rand) (*artifact.Artifact, error) {
-	buildPipeline := manifest.NewBuild(m, runner, repos, nil)
-	buildPipeline.Checkpoint()
-
-	osPipeline := manifest.NewOS(buildPipeline, img.Platform, repos)
-	osPipeline.PartitionTable = img.PartitionTable
-	osPipeline.OSCustomizations = img.OSCustomizations
-	osPipeline.Environment = img.Environment
-	osPipeline.Workload = img.Workload
-	osPipeline.NoBLS = img.NoBLS
-	osPipeline.OSProduct = img.OSProduct
-	osPipeline.OSVersion = img.OSVersion
-	osPipeline.OSNick = img.OSNick
-	if img.InstallWeakDeps != nil {
-		osPipeline.InstallWeakDeps = *img.InstallWeakDeps
+func makeImagePipeline(imgType platform.ImageFormat, rawImagePipeline manifest.FilePipeline, buildPipeline manifest.Build, opts *imagePipelineOpts) manifest.FilePipeline {
```

Comment:
Seems like for this type of stuff at least a single line doc would be helpful
```suggestion
// makeImagePipeline is a shared function to create a disk image of a desired type, including content from another pipeline (e.g. plain rpms, ostree, bootc, etc.)
func makeImagePipeline(imgType platform.ImageFormat, rawImagePipeline manifest.FilePipeline, buildPipeline manifest.Build, opts *imagePipelineOpts) manifest.FilePipeline {
```